en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/production-deployment-guideline.md

```diff
@@ -0,0 +1,288 @@
+# Production Deployment Guideline
+
+Given below are the common guidelines for deploying Choreo Connect in a **production environment** using [Helm deployment option](http://localhost:8000/deploy-and-publish/deploy-on-gateway/choreo-connect/getting-started/deploy/cc-deploy-overview/).
+
+## Configuring Keystore
+
+By default, Choreo Connect ships with default keystore certificates, and it is recommended to replace them with Self-Signed or CA signed certificates.
+
+### Setep 1: Generate Self-Signed certs in PEM format
+
+Generate Self-Signed certs in PEM format by executing following sample commands. Let's create a directory called `certs`.
+
+```shell
+mkdir -p certs/adapter certs/enforcer certs/router
+```
+
+```shell tab='Format'
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 \
+    -subj "/C=<COUNTRY_CODE>/ST=<STATE_NAME>/L=<LOCALITY_NAME>/O=<ORGANIZATION_NAME>/OU=<ORGANIZATION_UNIT_NAME>/CN=<COMMON_NAME>"\
+    -extensions SAN \
+    -config <(cat /etc/ssl/openssl.cnf \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:<DNS1>,DNS:<DNS2>")) \
+    -keyout <KEY_FILE_PATH> \
+    -out <CERT_FILE_PATH>
+```
+
+```shell tab='Adapter Sample'
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 \
+    -subj "/C=US/ST=CA/L=Mountain View/O=WSO2, Inc./OU=Choreo Connect/CN=yourdomain.com" \
+    -extensions SAN \
+    -config <(cat /etc/ssl/openssl.cnf \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:adapter")) \
+    -keyout certs/adapter/tls.key \
+    -out certs/adapter/tls.crt
+```
+
+```shell tab='Enforcer Sample'
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 \
+    -subj "/C=US/ST=CA/L=Mountain View/O=WSO2, Inc./OU=Choreo Connect/CN=yourdomain.com" \
+    -extensions SAN \
+    -config <(cat /etc/ssl/openssl.cnf \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:enforcer")) \
+    -keyout certs/enforcer/tls.key \
+    -out certs/enforcer/tls.crt
+```
+
+```shell tab='Router Sample'
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 \
+    -subj "/C=US/ST=CA/L=Mountain View/O=WSO2, Inc./OU=Choreo Connect/CN=yourdomain.com" \
+    -extensions SAN \
+    -config <(cat /etc/ssl/openssl.cnf \
+        <(printf "\n[SAN]\nsubjectAltName=DNS:router")) \
+    -keyout certs/router/tls.key \
+    -out certs/router/tls.crt
+```
+
+By default, SSL Hostnames for adapter and enforcer is `adapter` and `enforcer`, in case if you want to have your own SAN names in the certificates, update the following values in helm chart with the SAN names in the certificates.
```

Comment:
```suggestion
By default, SSL Hostnames for adapter and enforcer is `adapter` and `enforcer`, in case if you want to have your own SAN names in the certificates, update the following values in the Helm Chart with the SAN names in the certificates.
```